crates/sui-framework/sources/governance/validator.move

```diff
@@ -284,11 +284,10 @@ module sui::validator {
 
     /// Process pending delegations and withdraws, called at the end of the epoch.
     public(friend) fun process_pending_delegations_and_withdraws(self: &mut Validator, ctx: &mut TxContext) {
-        staking_pool::process_pending_delegations(&mut self.delegation_staking_pool, ctx);
         let reward_withdraw_amount = staking_pool::process_pending_delegation_withdraws(
             &mut self.delegation_staking_pool, ctx);
         self.metadata.next_epoch_delegation = self.metadata.next_epoch_delegation - reward_withdraw_amount;
-        // assert!(delegate_amount(self) == self.metadata.next_epoch_delegation, 0);
+        staking_pool::process_pending_delegations(&mut self.delegation_staking_pool, ctx);
```

Comment:
Do we want to add a TODO here so that one day we could add the assert back? Or is it never going to be guaranteed down the road?

Reply:
Will add a TODO here. I think this assert does hold today but I don't want to risk having it here and failing advance_epoch for next_epoch_delegation which doesn't matter too much now.